src/main/webapp/app/exercises/programming/manage/instructions-editor/analysis/programming-exercise-instruction-analysis.service.ts

```diff
@@ -152,21 +148,18 @@ export class ProgrammingExerciseInstructionAnalysisService {
      * @param regex to search for in the tasks.
      */
     private extractRegexFromTasks(tasks: [number, string][], regex: RegExp): [number, string[]][] {
-        return compose(
-            map(([lineNumber, match]) => {
-                const cleanedMatches = compose(
-                    uniq,
-                    filter((m) => !!m),
-                    flatten,
-                )(match.split(/,(?![^(]*?\))/).map((m: string) => m.trim()));
-                return [lineNumber, cleanedMatches];
-            }),
-            filter(([, testCases]) => !!testCases),
-            map(([lineNumber, task]) => {
+        const cleanMatches = (matches: string[]) => uniq(matches.flat().filter((m) => !!m));
```

Comment:
```suggestion
        const cleanMatches = (matches: string[]) => uniq(matches.flat().filter(Boolean);
```
I read it written like this somewhere, not sure if that would work here

Reply:
Thanks, converting to a boolean without need for the additional lambda looks nicer.